src/api/resources/VehicleEvents/index.ts

```diff
@@ -0,0 +1,196 @@
+/** @format */
+
+import { Method } from "api/Method";
+import { Resource } from "api/Resource";
+import { DimoEnvironment } from "environments";
+import { WebhookDefinitionParams } from "./types";
+
+const endpoints = {
+	listWebhooks: {
+		method: "GET",
+		path: "/v1/webhooks",
+		auth: "developer_jwt",
+	},
+	createWebhook: {
+		method: "POST",
+		path: "/v1/webhooks",
+		body: {
+			service: true,
+			data: true,
+			trigger: true,
+			setup: true,
+			description: false,
+			target_uri: true,
+			status: true,
+			verification_token: true,
+		},
+		auth: "developer_jwt",
+	},
+	updateWebhook: {
+		method: "PUT",
+		path: "/v1/webhooks/:webhookId",
+		body: {
+			service: true,
+			data: true,
+			trigger: true,
+			setup: true,
+			description: false,
+			target_uri: true,
+			status: true,
+			verification_token: true,
+		},
+		auth: "developer_jwt",
+	},
+	deleteWebhook: {
+		method: "DELETE",
+		path: "/v1/webhooks/:webhookId",
+		auth: "developer_jwt",
+	},
+	getWebhookSignalNames: {
+		method: "GET",
+		path: "/v1/webhooks/signals",
+		auth: "developer_jwt",
+	},
+	listSubscribedVehicles: {
+		method: "GET",
+		path: "/v1/webhooks/:webhookId",
+		auth: "developer_jwt",
+	},
+	listVehicleSubscriptions: {
+		method: "GET",
+		path: "/v1/webhooks/vehicles/:tokenId",
+		auth: "developer_jwt",
+	},
+	subscribeVehicle: {
+		method: "POST",
+		path: "/v1/webhooks/:webhookId/subscribe/:tokenId",
+		auth: "developer_jwt",
+	},
+	subscribeAllVehicles: {
+		method: "POST",
+		path: "/v1/webhooks/:webhookId/subscribe/all",
+		auth: "developer_jwt",
+	},
+	unsubscribeVehicle: {
+		method: "DELETE",
+		path: "/v1/webhooks/:webhookId/unsubscribe/:tokenId",
+		auth: "developer_jwt",
+	},
+	unsubscribeAllVehicles: {
+		method: "DELETE",
+		path: "/v1/webhooks/:webhookId/unsubscribe/all",
+		auth: "developer_jwt",
+	},
+};
+
+class VehicleEvents extends Resource {
+	constructor(api: any, env: keyof typeof DimoEnvironment) {
+		super(api, "VehicleEvents", env);
+	}
+
+	listWebhooks(authToken: string) {
```

Comment:
We actually call `setResource` inside the constructor to automagically setup the endpoints declared above instead of needing to redefine the functions. Check https://github.com/DIMO-Network/data-sdk/blob/master/src/api/resources/Valuations/index.ts

Reply:
@yusuf-cirak https://github.com/DIMO-Network/data-sdk/blob/a1f7babed90f3b6c46d872d6e8910769f87e3938/src/api/Resource.ts#L19

Reply:
Maybe you can improve the `setResource` function with a Resource interface type to help with type safety instead?